ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java

```diff
@@ -38,28 +37,51 @@
           + " as there are cores you want to use, as each instance is single threaded.";
 
   public static final String LISTEN_HOST = propertyName("listen.host");
-  public static final String DEFAULT_LISTEN_HOST = "0.0.0.0";
+  public static final String DEFAULT_LISTEN_HOST = "localhost";
   public static final String LISTEN_HOST_DOC =
       "The hostname to listen on";
 
   public static final String LISTEN_PORT = propertyName("listen.port");
-  public static final int DEFAULT_LISTEN_PORT = 8089;
+  public static final int DEFAULT_LISTEN_PORT = 8088;
   public static final String LISTEN_PORT_DOC =
       "The port to listen on";
 
-  public static final String KEY_PATH = propertyName("key.path");
-  public static final String KEY_PATH_DOC =
-      "Path to key file";
+  public static final String TLS_ENABLED = propertyName("tls.enabled");
+  public static final boolean DEFAULT_TLS_ENABLED = false;
+  public static final String TLS_ENABLED_DOC =
+      "Is TLS enabled?";
 
-  public static final String CERT_PATH = propertyName("cert.path");
-  public static final String CERT_PATH_DOC =
-      "Path to cert file";
+  public static final String TLS_KEY_STORE_PATH = propertyName("tls.keystore.path");
+  public static final String TLS_KEY_STORE_PATH_DOC =
+      "Path to server key store";
+
+  public static final String TLS_KEY_STORE_PASSWORD = propertyName("tls.keystore.password");
+  public static final String TLS_KEY_STORE_PASSWORD_DOC =
+      "Password for server key store";
+
+  public static final String TLS_TRUST_STORE_PATH = propertyName("tls.truststore.path");
+  public static final String TLS_TRUST_STORE_PATH_DOC =
+      "Path to client trust store";
+
+  public static final String TLS_TRUST_STORE_PASSWORD = propertyName("tls.truststore.password");
+  public static final String TLS_TRUST_STORE_PASSWORD_DOC =
+      "Password for client trust store";
+
+  public static final String TLS_CLIENT_AUTH_REQUIRED = propertyName("tls.client.auth.required");
+  public static final boolean DEFAULT_TLS_CLIENT_AUTH_REQUIRED = false;
+  public static final String TLS_CLIENT_AUTH_REQUIRED_DOC =
+      "Is client auth required?";
 
   public static final String WORKER_POOL_SIZE = propertyName("worker.pool.size");
   public static final String WORKER_POOL_DOC =
       "Max number of worker threads for executing blocking code";
   public static final int DEFAULT_WORKER_POOL_SIZE = 100;
 
+  public static final String MAX_PUSH_QUERIES = propertyName("max.push.queries");
+  public static final int DEFAULT_MAX_PUSH_QUERIES = 100;
```

Comment:
I'd say 100 is way too high!  Though, of course, it depends on _what_ queries they're running.  

@rodesai did some work investigating this, so he may have some more thoughts on what's a good limit.

Personally, I'd be starting with more like 10.

Reply:
+1 - when we benchmarked, we saw performance start to drop off around 30 queries (presumably due to overhead from all the threads added by each new streams app - streams threads, kafka client threads))

Reply:
I haven't tuned anything yet, so this is just a placeholder.

Reply:
FWIW - we were talking to someone who ran 415 (albeit super simple) queries on one node (we should verify), so might be better to just let the users tune this and start with something somewhat aggressive

Reply:
Erm.... I think it being a config allows the user to tune it @agavra :p lol.

Reply:
`-___-`